evm/contract.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package evm
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/utils"
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var ErrFailedReceiptStatus = fmt.Errorf("failed receipt status")
+
+func removeSurroundingParenthesis(s string) (string, error) {
+	s = strings.TrimSpace(s)
+	if len(s) > 0 {
+		if string(s[0]) != "(" || string(s[len(s)-1]) != ")" {
+			return "", fmt.Errorf("expected esp %q to be surrounded by parenthesis", s)
+		}
+		s = s[1 : len(s)-1]
+	}
+	return s, nil
+}
+
+func removeSurroundingBrackets(s string) (string, error) {
+	s = strings.TrimSpace(s)
+	if len(s) > 0 {
+		if string(s[0]) != "[" || string(s[len(s)-1]) != "]" {
+			return "", fmt.Errorf("expected esp %q to be surrounded by parenthesis", s)
+		}
+		s = s[1 : len(s)-1]
+	}
+	return s, nil
+}
+
+func getWords(s string) []string {
```

Comment:
if I understand this code correctly it can be done using regexp
like 
```
func getWords(s string) []string {
	pattern := `\([^()]*\)|\[[^\[\]]*\]|[^\s,]+`
	re := regexp.MustCompile(pattern)

	
	matches := re.FindAllString(s, -1)
	return matches
}
```

Reply:
also should be in utils

Reply:
this is pretty specific to evm, should not live in utils. will try to add unit test for example of what is doing.
I also prefer to avoid regex for this one, as regex big expressions tend to get harder to fix in many cases.
anyway will check the suggestion.

Reply:
the suggestion is not fully satisfying the unit tests and is complicated to debug and follow.
better will be for a posterior implementation for getWords to return a list of types and compounds types
where recursion is already handled